File: Python/p2.py
#Problem 1: even fibonacci numbers below 4m

def fibonacci(cap):
  fibos = [1,2]

  while fibos[-1] < cap:
    fibos.append (fibos[-1] + fibos[-2])

  return fibos

def even_fibos_sum():
  fibos = fibonacci(4000000)
  fibo_sum = 0

  for fibo in fibos:
    if fibo % 2 == 0:
      fibo_sum += fibo

  return fibo_sum

def main():
  print even_fibos_sum ()

if __name__ == '__main__':
  main()

File: Haskell/p2.hs
--P2 - Sum of even fibonacci below 4mill

p2 = sum $ takeWhile (<4000000) $ filter (even) (fibonacci 30 [2,1])

fibonacci 0 xs = xs
fibonacci n coso@(x:y:xs) = fibonacci (n-1) ((x + y) : coso)